src/services/UserService.ts

```diff
@@ -93,21 +77,170 @@ const getOutstandingOrdersCount = async(payload: any): Promise<any> => {
+      url: "getPermissions",
+      method: "post",
+      baseURL,
+      data: params,
+      headers: {
+        Authorization: 'Bearer ' + token,
+        'Content-Type': 'application/json'
+      }
+    })
+    if (resp.status === 200 && resp.data.docs?.length && !hasError(resp)) {
+      serverPermissions = resp.data.docs.map((permission: any) => permission.permissionId);
+      const total = resp.data.count;
+      const remainingPermissions = total - serverPermissions.length;
+      if (remainingPermissions > 0) {
+        // We need to get all the remaining permissions
+        const apiCallsNeeded = Math.floor(remainingPermissions / viewSize) + (remainingPermissions % viewSize != 0 ? 1 : 0);
+        const responses = await Promise.all([...Array(apiCallsNeeded).keys()].map(async (index: any) => {
+          const response = await client({
+            url: "getPermissions",
+            method: "post",
+            baseURL,
+            data: {
+              "viewIndex": index + 1,
+              viewSize,
+              permissionIds: payload.permissionIds
+            },
+            headers: {
+              Authorization: 'Bearer ' + token,
+              'Content-Type': 'application/json'
+            }
+          })
+          if (!hasError(response)) {
+            return Promise.resolve(response);
+          } else {
+            return Promise.reject(response);
+          }
+        }))
+        const permissionResponses = {
+          success: [],
+          failed: []
+        }
+        responses.reduce((permissionResponses: any, permissionResponse: any) => {
+          if (permissionResponse.status !== 200 || hasError(permissionResponse) || !permissionResponse.data?.docs) {
+            permissionResponses.failed.push(permissionResponse);
+          } else {
+            permissionResponses.success.push(permissionResponse);
+          }
+          return permissionResponses;
+        }, permissionResponses)
+
+        serverPermissions = permissionResponses.success.reduce((serverPermissions: any, response: any) => {
+          serverPermissions.push(...response.data.docs.map((permission: any) => permission.permissionId));
+          return serverPermissions;
+        }, serverPermissions)
+
+        // If partial permissions are received and we still allow user to login, some of the functionality might not work related to the permissions missed.
+        // Show toast to user intimiting about the failure
+        // Allow user to login
+        // TODO Implement Retry or improve experience with show in progress icon and allowing login only if all the data related to user profile is fetched.
+        if (permissionResponses.failed.length > 0) Promise.reject("Something went wrong while getting complete user permissions.");
+      }
+    }
+    return serverPermissions;
+  } catch (error: any) {
+    return Promise.reject(error);
+  }
+}
+
+const getUserProfile = async (token: any): Promise<any> => {
```

Comment:
If user is not associated with any facility we should not allow to login

Reply:
Updated sir